app/src/main/java/com/example/drawingactivity/DrawViewModel.kt

```diff
@@ -82,95 +87,120 @@ class DrawViewModel : ViewModel() {
     /**
      * Updates the viewModel with the status of open add dialog
      */
-    fun setDialogState() {
-        isDialogOpen = true
+    fun setAddDialogState() {
```

Comment:
Remember, the `View` tells the `ViewModel` about `EVENTS THAT HAPPEN`, nothing more. The view is to be dumb. Let's rename this with that in mind.

If a button click opens an add dialog, don't tell the viewmodel it needs to open the add dialog. That's letting the view logic know about business logic. Simply tell the viewmodel which button was clicked

Reply:
Renamed these to refer to how a dialog event is opened instead of setting a state.
example: `addDialogOpened()` instead of `setAddDialogState()`